File: app/src/main/java/au/csiro/ozatlas/model/SightingPhoto.java
package au.csiro.ozatlas.model;

import android.net.Uri;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import io.realm.RealmObject;

/**
 * Created by sad038 on 2/5/17.
 */

public class SightingPhoto extends RealmObject{
    public SightingPhoto() {
    }

    @Expose
    @SerializedName("dateTaken")
    public String dateTaken;
    @Expose
    @SerializedName("licence")
    public String licence;
    @Expose
    @SerializedName("name")
    public String name;
    @Expose
    @SerializedName("thumbnailUrl")
    public String thumbnailUrl;
    @Expose
    @SerializedName("filename")
    public String filename;
    @Expose
    @SerializedName("contentType")
    public String contentType;
    @Expose
    @SerializedName("staged")
    public Boolean staged;
    @Expose
    @SerializedName("notes")
    public String notes;
    @Expose
    @SerializedName("url")
    public String url;
    @Expose
    @SerializedName("attribution")
    public String attribution;
    @Expose
    @SerializedName("licensePosition")
    public int licensePosition;

    //Not part of Json
    public String filePath;
}

File: ozatlas/src/main/kotlin/au/org/ala/mobile/ozatlas/biocollect/upload/SightingPhoto.kt
package au.org.ala.mobile.ozatlas.biocollect.upload

import org.threeten.bp.Instant

data class SightingPhoto @JvmOverloads constructor (
    var attribution: String? = null,
    var contentType: String? = null,
    var dateTaken: Instant? = null,
    var documentId: String? = null,
    var filename: String? = null,
    var filesize: Int? = null,
    var formattedSize: String? = null,
    var licence: String? = null,
    var name: String? = null,
    var notes: String? = null,
    var staged: Boolean? = null,
    var status: String? = null,
    var thumbnailUrl: String? = null,
    var url: String? = null
)